bin/bovine.py

```diff
@@ -11,8 +11,9 @@ def __init__(self, root_directory):
     # get the files in the root_directory
     # self.
     self.root_directory = root_directory
+    self.filenames = filenames
     # import the yaml from the files that we found above
-  def list_files(self, root_directory):
+  def get_file_names(self, root_directory):
     directory_files = os.walk(root_directory)
     filenames = []
```

Comment:
in this function, there is no private var called filenames.  Remember, to access instance variables, you must use 'self.var_name'.